WordPress/gutenberg
# Fix Comments blocks pagination

## What
Make comments pagination links work as they should be when we have these conditions:
 - Comment Pagination enabled on Discussion Settings
 - Comment Query Loop "Inherit option" enabled (current status by default)

## Why
Both "Older Comments" and "Newer Comments" are not working as expected inside the `Comment Query Loop`.
If we enable pagination on Discussion Settings, before the PR we have:

- Comments are displayed in the correct order ✅
- Pagination is incorrect ❌

Older/Newer Comments URL, depending on what we have chosen on `Discussion Settings` (first or last page as the default for Comments Pagination), is not pointing to the correct URL. It makes comments pagination navigation inconsistent.

The reason for the bug is that we don't have the query var `cpage` defined and we are creating a `WP_Comment_Query( $comment_args )` with pagination arguments. In this case, the pagination link functions require this query var in order to know in which page of the comments pagination is the user when she lands on the default page/post view.

## How

Defining a default value for `cpage`, being the maximum number of pages or 0, depending on the Discussion setting chosen (show the first or the last page).

This way, when the user lands on a page/post without the comment pagination defined, it will land with the default value, so comments pagination links can work as expected.

## Testing Instructions
There are mainly two cases to test here:
1. Check that the "Older Comments" and "Newer Comments" link are shown correctly when the "Last page displayed by default" setting is enabled as explained in [this comment](https://github.com/WordPress/gutenberg/issues/37560#issuecomment-1042768856).
2. Check that the penultimate page links are working properly as explained in [this comment](https://github.com/WordPress/gutenberg/issues/37560#issuecomment-1059395812).

## Types of changes
Bug fix
